sound/soc/sof/sof-audio.c

```diff
@@ -180,6 +180,9 @@ void sof_widgets_free_all(struct snd_soc_component *scomp)
 	 */
 	list_for_each_entry_reverse(swidget, &sdev->widget_list, list)
 		sof_widget_free(sdev, swidget);
+
+	list_for_each_entry_reverse(swidget, &sdev->pipe_widget_list, list)
+		sof_widget_free(sdev, swidget);
```

Comment:
In the absence of comments it's hard to know if the widget_list and pipe_widget_list are independent and what this is used for....

Reply:
also this should probably be a separate patch for the ASoC core that explains what this is about, then a second SOF-specific one uses it.